src/core/action.ts

```diff
@@ -27,6 +27,12 @@ export function createAction(actionName: string, fn: Function, ref?: Object): Fu
         return executeAction(actionName, fn, ref || this, arguments)
     }
     ;(res as any).isMobxAction = true
+    if (process.env.NODE_ENV !== "production") {
```

Comment:
Implementing it this way will run the check every time an action is created, however, since this code might be hit hunderds of thousands times, and the result only depends on the environment, you only need to run it once. I suggest to pull this code up, run the check only once during module initialization, and reuse the result.

Reply:
(either `name` is configurable on all Functions, or none)

Reply:
also, could you add a unit test verifying this behavior?